src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -421,22 +421,32 @@ def workflow_notification(
 def build_comment_webhook(
     installation_id: int, issue_id: int, type: str, user_id: int, *args: Any, **kwargs: Any
 ) -> None:
-    webhook_data = get_webhook_data(installation_id, issue_id, user_id)
-    install, _, user = webhook_data
-    data = kwargs.get("data", {})
-    project_slug = data.get("project_slug")
-    comment_id = data.get("comment_id")
-    payload = {
-        "comment_id": data.get("comment_id"),
-        "issue_id": issue_id,
-        "project_slug": data.get("project_slug"),
-        "timestamp": data.get("timestamp"),
-        "comment": data.get("comment"),
-    }
-    send_webhooks(installation=install, event=type, data=payload, actor=user)
-    # `type` is comment.created, comment.updated, or comment.deleted
+    try:
+        event = SentryAppEventType(type)
+    except ValueError as e:
+        raise SentryAppSentryError(message=SentryAppWebhookFailureReason.INVALID_EVENT) from e
+
+    with SentryAppInteractionEvent(
+        operation_type=SentryAppInteractionType.PREPARE_WEBHOOK,
+        event_type=event,
+    ).capture():
```

Comment:
Do we want any additional lifecycle data here to log if this raises? Things like sentry app ID, installation ID, issue ID for example?

Reply:
We're not adding extra logging context currently as the info we would've added to lifecycle, would be the already captured in sentry from the function parameters. Idk if there's an advantage one way or the other, ig our lifecycle logs become more useful and can cross reference easier (if we added extra context). As opposed to not having redundant data (if we didn't add extra context), though at the same time the info captured by sentry would p. much be the same as the log 🤷‍♀️ 

- I can add whilst doing the other edit

Reply:
Having the installation ID would be helpful just for us to filter logs when debugging failures, but it's your call. We can always add these later if we find that the default logging is insufficient so don't worry to much about it.